contracts/competition/Competition.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "../referrals/interfaces/IReferralStorage.sol";
+import "../access/Governable.sol";
+
+contract Competition is Governable
+{
+    struct Team {
+        address leader;
+        string name;
+        bytes32 referral;
+        address[] members;
+        address[] joinRequests;
+    }
+
+    struct CompetitionDetails {
+        uint start;
+        uint end;
+        uint maxTeamSize;
+    }
+
+    CompetitionDetails public competitionDetails;
```

Comment:
mm i think X meant to add that struct to support multiple competitions inside a contract.

so we need to be able to create new competitions, update/delete them (with some validations, e.g. it should be impossible to delete ongoing competitions i guess. or set start date < now)